cdc/api/middleware/middleware.go

```diff
@@ -185,3 +187,56 @@ func CheckServerReadyMiddleware(capture capture.Capture) gin.HandlerFunc {
 		}
 	}
 }
+
+// AuthenticateMiddleware authenticates the request by query upstream TiDB.
+func AuthenticateMiddleware(capture capture.Capture) gin.HandlerFunc {
+	return func(ctx *gin.Context) {
+		m, err := capture.GetUpstreamManager()
+		if err != nil {
+			_ = ctx.Error(err)
+			ctx.Abort()
+			return
+		}
+		up, err := m.GetDefaultUpstream()
+		if err != nil {
+			_ = ctx.Error(err)
+			ctx.Abort()
+			return
+		}
+
+		if err := verify(ctx, up); err != nil {
+			ctx.IndentedJSON(http.StatusUnauthorized, model.NewHTTPError(err))
+			ctx.Abort()
+			return
+		}
+		ctx.Next()
+	}
+}
+
+func verify(ctx *gin.Context, up *upstream.Upstream) error {
+	serverCfg := config.GetGlobalServerConfig()
+	if serverCfg.Security.ClientUserRequired {
+		username := ctx.Query(api.ApiOpVarTiCDCUser)
```

Comment:
specify username& password in URL is not recommended, it's better to use the `Authorization` header